index.html

```diff
@@ -221,7 +221,8 @@ <h2>
       </ol>
       <p>
         To <dfn>obtain permission</dfn> for <a>wake lock type</a>
-        |type|, run these steps <a>in parallel</a>:
+        |type|, run these steps <a>in parallel</a>. This async algorithm
+        returns either "<a>granted</a>" or "<a>denied</a>".
       </p>
       <ol class="algorithm">
         <li>
```

Comment:
Below this line, there is a PermissionDescriptor being created inside a `pre`... that's not correct. You need:

> Let |permissionDesc:PermissionDescriptor| be the following be a newly created  PermissionDescriptor.
> Set |permissionDesc|.<a data-link-for="PermissionDescriptor">name</a> to "wake-lock",
> Set |permissionDesc|.<a data-link-for="PermissionDescriptor">type</a> to | type|.

Reply:
This followed the approach used in Web USB. I can change of course, but @reillyeon input might be useful

Reply:
Because we got here from `requestPermission()`,  we are already in WebIDL land inside the algorithm (so technically in the C++ implementation)... so you can literally just create: `PermissionDescriptor permissionDesc;` and then assign it the name (as above).

Reply:
Maybe you could file an issue for Web USB?

Reply:
Done